blocks/api/raw-handling/index.js

```diff
@@ -46,16 +46,30 @@ export default function rawHandler( { HTML, plainText = '', mode = 'AUTO', tagNa
 		return parseWithGrammar( HTML );
 	}
 
-	// If there is a plain text version, the HTML version has no formatting,
-	// and there is at least a double line break,
-	// parse any Markdown inside the plain text.
-	if ( plainText && isPlain( HTML ) && plainText.indexOf( '\n\n' ) !== -1 ) {
+	// Parse Markdown (and HTML) if:
+	// * There is a plain text version.
+	// * The HTML version has no formatting.
+	if ( plainText && isPlain( HTML ) ) {
 		const converter = new showdown.Converter();
 
 		converter.setOption( 'noHeaderId', true );
 		converter.setOption( 'tables', true );
 
 		HTML = converter.makeHtml( plainText );
+
+		// Switch to inline mode if:
+		// * The current mode is AUTO.
+		// * The original plain text had no line breaks.
+		// * The original plain text was not an HTML paragraph.
+		// * The converted text is just a paragraph.
+		if (
+			mode === 'AUTO' &&
+			plainText.indexOf( '\n' ) === -1 &&
+			plainText.indexOf( '<p>' ) === -1 &&
```

Comment:
Strictly speaking, this line doesn't correspond to its English-language counterpart:

> The original plain text was not an HTML paragraph.

That would be coded as `plainText.indexOf( '<p>' ) !== 0`, right?

Furthermore, it made me think of special cases, as illustrated by this breaking test:

```js
it( 'should parse Markdown with backtick-escaped tags', () => {
	const filtered = rawHandler( {
		HTML: '* one<br>* two `<p>`<br>* three',
		plainText: '* one\n* two\n* three',
		mode: 'AUTO',
	} ).map( getBlockContent ).join( '' );

	equal( filtered, '<ul>\n    <li>one</li>\n    <li>two <code>&lt;p&gt;</code></li>\n    <li>three</li>\n</ul>' );
} );
```

Reply:
@mcsf Normally that should be the following and pass:

```js
it( 'should parse Markdown with backtick-escaped tags', () => {
	const filtered = rawHandler( {
		HTML: '* one<br>* two `&lt;p&gt;`<br>* three',
		plainText: '* one\n* two `<p>`\n* three',
		mode: 'AUTO',
	} ).map( getBlockContent ).join( '' );

	equal( filtered, '<ul>\n    <li>one</li>\n    <li>two <code>&lt;p&gt;</code></li>\n    <li>three</li>\n</ul>' );
} );
```

I tried this in Safari and it works. Now I also tried that is Chrome, and it does not encode HTML tags from from plain text to the HTML version, which breaks `isPlain`... 🤷‍♀️

Reply:
In other words, I'm not sure if we can check if paste is from plain text in Chrome?

Reply:
Safari:

```
[Log] Received HTML: (index.js, line 2790)

 – "* one<br>* two `&lt;p&gt;`<br>* three<br>"
[Log] Received plain text: (index.js, line 2791)

 – "* one↵* two `<p>`↵* three↵"
"* one
* two `<p>`
* three
"
[Log] Processed HTML piece: (index.js, line 31264)

 – "<ul>↵<li>one</li>↵<li>two <code>&lt;p&gt;</code></li>↵<li>three</li>↵</ul>"
"<ul>
<li>one</li>
<li>two <code>&lt;p&gt;</code></li>
<li>three</li>
</ul>"
```

Chrome:

```
index.js:316 Received HTML:

 * one
* two `<p>`
* three

index.js:317 Received plain text:

 * one
* two `<p>`
* three

index.js:133 Processed HTML piece:

 <p>* one
* two `</p><p>`
* three
</p>
```

Reply:
Addressed the `plainText.indexOf( '<p>' ) !== 0` bit.

Reply:
This will also be an issue in Chrome if there is any HTML at all in the Markdown.

```md
* one
* two <em>test</em>
* three
```

Will work in Safari but fail in Chrome.

Reply:
I opened #4256, since it is not really related to what this PR is trying to fix.

Reply:
Aha, I hadn't compared with Safari (also, yes, forgot to update `plainText` in my example). Thanks for opening a new issue.